### PR TITLE
Improve `configurable` variable support for basic types

### LINKED ISSUE #27129: [Configurable] Migrate the configuration Toml parsing to use ProjectAPI TomlParser
**Description:**
Currently, we use a custom toml parser to read the `configuration.toml` file. We need improve this by using the already implemented parser used by Project API to avoid duplication. 

**Steps to reproduce:**

**Affected Versions:**
SLP6

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**
Team/jBallerina, Language/Configurable

**Suggested Assignees (optional):**

### LINKED ISSUE #27131: [Configurable] Improve configuration support for other types
**Description:**
Currently we support only `int`, `float`,`boolean` and `string` type variables as configurables. We need to support the other types as well.

Further related improvements - 
 - support optional organization name in TOML
> The organization name would only be required when necessary to resolve an ambiguity.
 - support toml key-value specification for directly specify module & variable name
   Eg. : 
```toml
[moduleName.subModuleName]
variable1 = 11
variable2 = "abc"
```
This should be equivalent to the following as well. 
```toml
moduleName.subModuleName.variable1 = 11
moduleName.subModuleName.variable2 = "abc"

```
**Steps to reproduce:**

**Affected Versions:**
SLP6

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**
Team/jBallerina, Language/Configurable

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

Fixes #27131 
Fixes #27129 

## Approach
-  The parsing and map population methods are updated with the new [TOML parser](https://github.com/ballerina-platform/ballerina-lang/tree/master/misc/toml-parser) .
- Added validations and value creation methods for `decimal` and array of basic types.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-rt/build.gradle [hunk 1]
```diff
@@ -84,6 +84,7 @@ dependencies {
     dist 'org.slf4j:slf4j-api:1.7.22'
     dist 'com.moandjiezana.toml:toml4j:0.7.2'
```
Comment (sameerajayasoma) [inline bvm/ballerina-rt/build.gradle:85]: Can we remove the `com.moandjiezana.toml:toml4j:0.7.2` dependency with this change?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigToml.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.List;
+
+import static io.ballerina.runtime.internal.configurable.ConfigurableConstants.INVALID_TOML_FILE;
+
+/**
+ * Represents configuration TOML document for `configurable` variables.
+ *
+ * @since 2.0.0
+ */
+public class ConfigToml {
```
Comment (sameerajayasoma) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigToml.java:44]: As explained by Hinduja, this is a duplication of the following class.
https://github.com/ballerina-platform/ballerina-lang/blob/6aa21902043111c5776679afdf28fafd071c8aa2/compiler/ballerina-lang/src/main/java/io/ballerina/projects/TomlDocument.java#L36

@HindujaB Can you please create an issue to fix this? We need to bring this class to some other module.

Comment (HindujaB) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigToml.java:44]: Created issue #27475

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/TomlParser.java [hunk 3]
```diff
@@ -47,88 +64,171 @@
...
+            switch (elementType.getTag()) {
+                case TypeTags.INT_TAG:
+                    checkTypeAndthrowError(TomlType.INTEGER, retrievedType, variableName, elementType);
+                    for (int i = 0; i < arraySize; i++) {
+                        arrayEntries[i] = new ListInitialValueEntry.ExpressionEntry(
+                                ((TomlLongValueNode) arrayList.get(i)).getValue());
+                    }
+                    break;
+                case TypeTags.BOOLEAN_TAG:
+                    checkTypeAndthrowError(TomlType.BOOLEAN, retrievedType, variableName, elementType);
```
Comment (manuranga) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/TomlParser.java:169]: can we replace with 
```java
createArray.<TomlType.BOOLEAN>(...)
...
public static <T> ListInitialValueEntry.ExpressionEntry[] createArray(....) {
        T rv = (T)o;
}
```

**Comment by warunalakshitha** [comment]
LGTM

**Review by manuranga — APPROVED** [review]

**Pull request merged**